units/migrate_parts.py

```diff
@@ -0,0 +1,43 @@
+import json
+import sys
+
+master_array = []
+
+def main():
+    with open("../master_part_list_2.csv", "r") as parts_file:
+        lines = parts_file.readlines()
+        first_line = True
+        for line in lines:
+            if first_line:
+                first_line = False
+                continue
+            line = line.rstrip('\n')
+            parseLine(line)
+
+    json_object = json.dumps(master_array, indent = 4)
+    print(json_object)
+    with open("../master_part_list.json", "w") as json_file:
+        json_file.write(json_object)
```

Comment:
instead of separately dumping on line 17 and then writing it on line 20, you could do:
```python
with open(...) as json_file:
    json_file.dump(master_array, indent=4)
```

You could even technically make this more efficient by converting the earlier data parsing into a generator where `parseLine` is yielded allowing processing a really huge files (not that we'd have files that large):
```python
def parseFile(parts_file):
   first_line = parts_file.readline() # drop the first line
   for line in parts_file.readlines(): # everything else
       yield parseLine(line.rstrip('\n'))

with open("../master_part_list.json", "w") as json_file:
    with open("../master_part_list_2.csv", "r") as parts_file:
        json_file.dump(parseFile(parts_file), indent=4)
```

Reply:
There's really no need to go that far. It's a one off script for a specific file. It doesn't get run every time by the game.

Reply:
true - if the file was super large (GB in size) then you'd have to just to keep it within memory boundaries.
As noted, with the change it's good enough. Thanks!